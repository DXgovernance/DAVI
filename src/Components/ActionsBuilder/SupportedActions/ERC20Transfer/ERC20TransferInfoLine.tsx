import { ActionViewProps } from '..';
import { Segment } from '../common/infoLine';
import { BigNumber } from 'ethers';
import useBigNumberToNumber from 'hooks/Guilds/conversions/useBigNumberToNumber';
import useENSAvatar from 'hooks/Guilds/ether-swr/ens/useENSAvatar';
import { useERC20Info } from 'hooks/Guilds/ether-swr/erc20/useERC20Info';
import Avatar from 'old-components/Guilds/Avatar';
import { useMemo } from 'react';
import { FiArrowRight, FiNavigation } from 'react-icons/fi';
import {
  getNetworkById,
  MAINNET_ID,
  shortenAddress,
  ZERO_ADDRESS,
} from 'utils';
import { useTranslation } from 'react-i18next';
import { useNetwork } from 'wagmi';

const ERC20TransferInfoLine: React.FC<ActionViewProps> = ({
  decodedCall,
  compact,
}) => {
  const { t } = useTranslation();
  const { chain } = useNetwork();

  const parsedData = useMemo(() => {
    if (!decodedCall) return null;

    const isNativeToken =
      decodedCall.value !== null && decodedCall.value._hex !== '0x00'
        ? true
        : false;

    if (isNativeToken) {
      return {
        tokenAddress: ZERO_ADDRESS,
        amount: BigNumber.from(decodedCall.value),
        source: decodedCall.from,
        destination: decodedCall.to as string,
      };
    } else {
      return {
        tokenAddress: decodedCall.to,
        amount: BigNumber.from(decodedCall.args._value),
        source: decodedCall.from,
        destination: decodedCall.args._to as string,
      };
    }
  }, [decodedCall]);

  const { data: erc20TokenInfo } = useERC20Info(parsedData?.tokenAddress);
  const { nativeAsset } = getNetworkById(chain?.id);

  let [tokenSymbol, tokenDecimals] = useMemo(() => {
    if (erc20TokenInfo !== undefined) {
      return [erc20TokenInfo.symbol, erc20TokenInfo.decimals];
    } else {
      return [nativeAsset.symbol, nativeAsset.decimals];
    }
  }, [erc20TokenInfo, nativeAsset]);

  const roundedBalance = useBigNumberToNumber(
    parsedData?.amount,
    tokenDecimals,
    4
  );
  const { ensName, imageUrl } = useENSAvatar(
    parsedData?.destination,
    MAINNET_ID
  );

  return (
    <>
      <Segment>
        <FiNavigation size={16} />
      </Segment>
      <Segment>
        {!compact ? t('transfer') : ''} {roundedBalance} {tokenSymbol}
      </Segment>
      <Segment>
        <FiArrowRight />
      </Segment>
      <Segment>
        <Avatar
          defaultSeed={parsedData?.destination}
          src={imageUrl}
          size={compact ? 14 : 24}
        />
      </Segment>
      <Segment>
        {ensName ||
          (parsedData?.destination
            ? shortenAddress(parsedData?.destination, compact ? 2 : 4)
            : '')}
      </Segment>
    </>
  );
};

export default ERC20TransferInfoLine;
