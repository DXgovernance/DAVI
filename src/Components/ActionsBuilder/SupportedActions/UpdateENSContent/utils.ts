import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
