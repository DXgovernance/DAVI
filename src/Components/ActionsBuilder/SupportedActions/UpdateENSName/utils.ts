import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
