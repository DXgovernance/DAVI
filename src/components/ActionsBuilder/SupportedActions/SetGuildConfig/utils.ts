import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
