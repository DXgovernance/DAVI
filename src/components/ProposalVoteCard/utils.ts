import { ConfirmVoteProposalProps, VoteOnProposalProps } from './types';

export const voteOnProposal = ({
  votingPowerAtProposalSnapshot,
  votingPowerAtProposalCurrentSnapshot,
}: VoteOnProposalProps) => {
  const hasNoVotingPower =
    votingPowerAtProposalSnapshot &&
    Number(votingPowerAtProposalSnapshot?.toString()) <= 0;

  const hasVotingPowerAtCurrentSnapshot =
    votingPowerAtProposalCurrentSnapshot &&
    Number(votingPowerAtProposalCurrentSnapshot?.toString()) > 0;
  return {
    hasNoVotingPower,
    hasVotingPowerAtCurrentSnapshot,
  };
};

export const confirmVoteProposal = ({
  proposal,
  contract,
  selectedOption,
  userVotingPower,
  createTransaction,
  cb,
}: ConfirmVoteProposalProps) => {
  createTransaction(
    `Vote on proposal ${proposal?.title}`,
    async () => contract.setVote(proposal?.id, selectedOption, userVotingPower),
    true,
    cb
  );
};

export const getOptionLabel = ({ metadata, optionKey, t }) => {
  const metadataLabel = metadata?.voteOptions?.[optionKey];
  return metadataLabel
    ? metadataLabel
    : Number(optionKey) === 0
    ? t('against', { defaultValue: 'Against' })
    : t('option', { optionKey });
};
