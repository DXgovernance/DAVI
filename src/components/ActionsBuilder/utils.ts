import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
