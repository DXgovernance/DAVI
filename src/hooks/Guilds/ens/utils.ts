// import { isSupportedChainId } from 'components/ActionsBuilder/SupportedActions/UpdateENSContent/utils';
import { MAINNET_ID } from 'utils';

export const isAvailableOnENS = (chainId: number) => {
  const ensNetworks = [MAINNET_ID];
  return ensNetworks.includes(chainId);
};
