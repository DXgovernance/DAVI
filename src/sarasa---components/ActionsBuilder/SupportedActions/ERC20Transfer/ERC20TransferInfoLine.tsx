import { ActionViewProps } from '..';
import { Segment } from '../common/infoLine';
import { BigNumber } from 'ethers';
import useBigNumberToNumber from 'hooks/Guilds/conversions/useBigNumberToNumber';
import useENSAvatar from 'hooks/Guilds/ether-swr/ens/useENSAvatar';
import { useERC20Info } from 'hooks/Guilds/ether-swr/erc20/useERC20Info';
import Avatar from 'components/Avatar';
import { useMemo } from 'react';
import { FiArrowRight } from 'react-icons/fi';
import { BiSend } from 'react-icons/bi';
import { MAINNET_ID, shortenAddress } from 'utils';
import { useTranslation } from 'react-i18next';

const ERC20TransferInfoLine: React.FC<ActionViewProps> = ({
  decodedCall,
  compact,
  noAvatar,
}) => {
  const { t } = useTranslation();

  const parsedData = useMemo(() => {
    if (!decodedCall) return null;

    return {
      tokenAddress: decodedCall.to,
      amount: BigNumber.from(decodedCall.args._value),
      source: decodedCall.from,
      destination: decodedCall.args._to as string,
    };
  }, [decodedCall]);

  const { data: tokenInfo } = useERC20Info(parsedData?.tokenAddress);
  const roundedBalance = useBigNumberToNumber(
    parsedData?.amount,
    tokenInfo?.decimals,
    4
  );
  const { ensName, imageUrl } = useENSAvatar(
    parsedData?.destination,
    MAINNET_ID
  );

  return (
    <>
      <Segment>
        <BiSend size={16} />
      </Segment>
      <Segment>
        {!compact ? t('transfer') : ''} {roundedBalance} {tokenInfo?.symbol}
      </Segment>
      <Segment>
        <FiArrowRight />
      </Segment>
      {noAvatar ? null : (
        <Segment>
          <Avatar
            defaultSeed={parsedData?.destination}
            src={imageUrl}
            size={compact ? 14 : 24}
          />
        </Segment>
      )}
      <Segment>
        {ensName ||
          (parsedData?.destination
            ? shortenAddress(parsedData?.destination, compact ? 2 : 4)
            : '')}
      </Segment>
    </>
  );
};

export default ERC20TransferInfoLine;
