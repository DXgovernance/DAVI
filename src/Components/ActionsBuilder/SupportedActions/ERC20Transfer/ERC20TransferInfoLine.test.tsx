import { BigNumber } from 'ethers';
import { ZERO_ADDRESS } from 'utils';
import { render } from 'utils/tests';
import ERC20TransferInfoLine from './ERC20TransferInfoLine';
import {
  erc20TransferDecodedCallMock,
  erc20TransferEmptyDecodedCallMock,
} from './fixtures';

const mockBigNumber = BigNumber.from(100000000);

jest.mock('hooks/Guilds/ether-swr/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

jest.mock('hooks/Guilds/ether-swr/erc20/useERC20Info', () => ({
  useERC20Info: () => ({
    name: 'Test ERC20',
    symbol: 'TEST',
    decimals: 18,
    totalSupply: mockBigNumber,
  }),
}));

const mockAddress = ZERO_ADDRESS;
const mockChainId = 123456;
jest.mock('wagmi', () => ({
  useAccount: () => ({ address: mockAddress }),
  useNetwork: () => ({ chain: { id: mockChainId } }),
}));

jest.mock('utils', () => ({
  getNetworkById: () => ({ nativeAsset: { symbol: 'ETH' } }),
}));

describe('ERC20TransferInfoLine', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <ERC20TransferInfoLine decodedCall={erc20TransferDecodedCallMock} />
    );
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshot in compact mode', () => {
    const { container } = render(
      <ERC20TransferInfoLine
        decodedCall={erc20TransferDecodedCallMock}
        compact={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshot without data', () => {
    const { container } = render(<ERC20TransferInfoLine decodedCall={null} />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot without data in compact mode', () => {
    const { container } = render(
      <ERC20TransferInfoLine decodedCall={null} compact={true} />
    );
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshot with default values', () => {
    const { container } = render(
      <ERC20TransferInfoLine decodedCall={erc20TransferEmptyDecodedCallMock} />
    );
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshot with default values in compact mode', () => {
    const { container } = render(
      <ERC20TransferInfoLine
        decodedCall={erc20TransferEmptyDecodedCallMock}
        compact={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
