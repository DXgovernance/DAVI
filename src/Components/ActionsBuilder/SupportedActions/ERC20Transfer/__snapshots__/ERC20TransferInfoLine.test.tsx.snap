// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ERC20TransferInfoLine Should match snapshot 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    transfer
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT kIpdaa"
      data-testid="avatar"
      size="24"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;

exports[`ERC20TransferInfoLine Should match snapshot in compact mode 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT iRVGHC"
      data-testid="avatar"
      size="14"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;

exports[`ERC20TransferInfoLine Should match snapshot with default values 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    transfer
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT kIpdaa"
      data-testid="avatar"
      size="24"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;

exports[`ERC20TransferInfoLine Should match snapshot with default values in compact mode 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT iRVGHC"
      data-testid="avatar"
      size="14"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;

exports[`ERC20TransferInfoLine Should match snapshot without data 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    transfer
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT kIpdaa"
      data-testid="avatar"
      size="24"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;

exports[`ERC20TransferInfoLine Should match snapshot without data in compact mode 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="3 11 22 2 13 21 11 13 3 11"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    
     
     
    ETH
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-gKseQn sc-iBPTik lGPNT iRVGHC"
      data-testid="avatar"
      size="14"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;
